### PR TITLE
fix(db & connection): make to show/hide the password when only creating db connection

### PR DESCRIPTION
**Pull request by prosdev0107**
### SUMMARY
Show/Hide Password in Database Connection Creation Dynamic Forms

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:
image

AFTER:
Screen Shot 2022-04-13 at 7 53 46 AM
Screen Shot 2022-04-13 at 7 53 57 AM

### TESTING INSTRUCTIONS
**Proposed changes:**

1. By default, anything typed in the password field should be masked
2. User should be able to toggle to see the password and toggle back to hiding the password ONLY WHEN entering the password when CONNECTING a database, not when EDITING one
3. When editing an existing db connection the password should be masked WITHOUT the option to show the password to prevent unintentionally showing passwords to someone who didn't initially connect the db
4. See details in comment on show/hide icon and tooltip text

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by AAfghahi — CHANGES_REQUESTED** [review]

File: superset-frontend/src/components/Form/LabeledErrorBoundInput.tsx [hunk 1]
```diff
@@ -17,7 +17,8 @@
  * under the License.
  */
 import React from 'react';
-import { Input } from 'antd';
+import { Input, Tooltip } from 'antd';
```
[the remark this answers is not in the document]
Comment (AAfghahi) [reply superset-frontend/src/components/Form/LabeledErrorBoundInput.tsx:20]: Hey I am sorry I was bugging and didn't realize this was the component itself. Sorry!

**Review by AAfghahi — APPROVED** [review]

**Review by codemaster08240328 — APPROVED** [review]
LGTM

**Review by diegomedina248 — APPROVED** [review]

**Pull request merged**